plugin/trino-mariadb/src/main/java/io/trino/plugin/mariadb/MariaDbClient.java

```diff
@@ -623,6 +647,102 @@ protected boolean isSupportedJoinCondition(ConnectorSession session, JdbcJoinCon
+                rowCount = null;
+            }
+
+            if (rowCount == null && indexMaxCardinality == null) {
+                // Table not found, or is a view, or has no usable statistics
+                return TableStatistics.empty();
+            }
+            rowCount = max(firstNonNull(rowCount, 0L), firstNonNull(indexMaxCardinality, 0L));
+
+            TableStatistics.Builder tableStatistics = TableStatistics.builder();
+            tableStatistics.setRowCount(Estimate.of(rowCount));
+
+            // TODO statistics from ANALYZE TABLE (https://mariadb.com/kb/en/engine-independent-table-statistics/)
+            // Map<String, AnalyzeColumnStatistics> columnStatistics = statisticsDao.getColumnStatistics(table);
+            Map<String, AnalyzeColumnStatistics> columnStatistics = ImmutableMap.of();
+
+            // TODO add support for histograms https://mariadb.com/kb/en/histogram-based-statistics/
+
+            // statistics based on existing indexes
+            Map<String, ColumnIndexStatistics> columnStatisticsFromIndexes = statisticsDao.getColumnIndexStatistics(table);
+
+            if (columnStatistics.isEmpty() && columnStatisticsFromIndexes.isEmpty()) {
+                log.debug("No column and index statistics read");
+                // No more information to work on
+                return tableStatistics.build();
+            }
+
+            for (JdbcColumnHandle column : getColumns(session, table)) {
+                ColumnStatistics.Builder columnStatisticsBuilder = ColumnStatistics.builder();
+
+                String columnName = column.getColumnName();
+                AnalyzeColumnStatistics analyzeColumnStatistics = columnStatistics.get(columnName);
+                if (analyzeColumnStatistics != null) {
+                    log.debug("Reading column statistics for %s, %s from analayze's column statistics: %s", table, columnName, analyzeColumnStatistics);
+                    columnStatisticsBuilder.setNullsFraction(Estimate.of(analyzeColumnStatistics.nullsRatio()));
+                }
+
+                ColumnIndexStatistics columnIndexStatistics = columnStatisticsFromIndexes.get(columnName);
+                if (columnIndexStatistics != null) {
+                    log.debug("Reading column statistics for %s, %s from index statistics: %s", table, columnName, columnIndexStatistics);
+                    columnStatisticsBuilder.setDistinctValuesCount(Estimate.of(columnIndexStatistics.cardinality()));
+
+                    if (!columnIndexStatistics.nullable()) {
+                        double knownNullFraction = columnStatisticsBuilder.build().getNullsFraction().getValue();
+                        if (knownNullFraction > 0) {
+                            log.warn("Inconsistent statistics, null fraction for a column %s, %s, that is not nullable according to index statistics: %s", table, columnName, knownNullFraction);
```

Comment:
did you mean "column %s.%s"? otherwise it would log `foo, bar` where `foo` is table and `bar` is column.

Reply:
i admit having copied this from MySQL's `io.trino.plugin.mysql.MySqlClient#updateColumnStatisticsFromIndexStatistics`

yes, we can improve both places

Reply:
I'll take care of this.